sympy/integrals/deltafunctions.py

```diff
@@ -172,5 +168,9 @@ def deltaintegrate(f, x):
                     dg, rest_mult_2 = change_mul(dg, x)
                     rest_mult = rest_mult*rest_mult_2
                 point = solve(dg.args[0], x)[0]
-                return (rest_mult.subs(x, point)*Heaviside(x - point))
+                try:
+                    n = dg.args[1]
+                except:
```

Comment:
I see. dg is a DiracDelta term, and DiracDelta(x).args is just (x,). I think we should fix it so that DiracDelta always has two args. Right now DiracDelta(x, 0) isn't even the same as DiracDelta(x). 

Even without that, the correct thing to do here is to check `len(dg.args)`, not to use a try/except.